tracer/build/_build/Build.ExplorationTests.cs

```diff
@@ -0,0 +1,301 @@
+                GitRepositoryTag = "v3.19.1",
+                PathToUnitTestProject = "csharp/src/Google.Protobuf.Test",
+                SupportedFrameworks = new[] { TargetFramework.NETCOREAPP2_1, TargetFramework.NET5_0, }
+            };
+        }
+
+        if (testName.Equals(ExplorationTestName.cake))
+        {
+            return new ExplorationTestDescription()
+            {
+                GitRepositoryUrl = "https://github.com/cake-build/cake.git",
+                Name = "cake",
+                GitRepositoryTag = "v1.3.0",
+                PathToUnitTestProject = "src/Cake.Common.Tests",
+                SupportedFrameworks = new[] { TargetFramework.NETCOREAPP3_1, TargetFramework.NET5_0, TargetFramework.NET6_0 }
+
+            };
+        }
+
+        if (testName.Equals(ExplorationTestName.Swashbuckle))
+        {
+            return new ExplorationTestDescription()
+            {
+                GitRepositoryUrl = "https://github.com/domaindrivendev/Swashbuckle.AspNetCore.git",
+                Name = "Swashbuckle.AspNetCore",
+                GitRepositoryTag = "v6.2.3",
+                PathToUnitTestProject = "test/Swashbuckle.AspNetCore.SwaggerGen.Test",
+                SupportedFrameworks = new[] { TargetFramework.NET6_0 }
+            };
+        }
+
+        if (testName.Equals(ExplorationTestName.Paket))
+        {
+            return new ExplorationTestDescription()
+            {
+                GitRepositoryUrl = "https://github.com/fsprojects/Paket.git",
+                Name = "Paket",
+                GitRepositoryTag = "6.2.1",
+                PathToUnitTestProject = "tests/Paket.Tests",
+                TestsToIgnore = new[] { "Loading assembly metadata works" },
+                SupportedFrameworks = new[] { TargetFramework.NET461, TargetFramework.NETCOREAPP3_1 }
+            };
+        }
+
+        if (testName.Equals(ExplorationTestName.ILSpy))
+        {
+            return new ExplorationTestDescription()
+            {
+                GitRepositoryUrl = "https://github.com/icsharpcode/ILSpy.git",
+                Name = "ILSpy",
+                GitRepositoryTag = "6.2.1",
+                PathToUnitTestProject = "ICSharpCode.Decompiler.Tests",
+                TestsToIgnore = new[] { "ICSharpCode.Decompiler.Tests", "UseMc", "_net45", "ImplicitConversions", "ExplicitConversions", "ICSharpCode_Decompiler", "NewtonsoftJson_pcl_debug", "NRefactory_CSharp", "Random_TestCase_1" },
+                SupportedFrameworks = new[] { TargetFramework.NET461, TargetFramework.NETCOREAPP3_1 }
+            };
+        }
+
+        throw new NotImplementedException($"This exploration test '{testName}' is not supported");
+    }
+}
+
+partial class Build
+{
+    AbsolutePath ExplorationTestsDirectory => RootDirectory / "exploration-tests";
+
+    [Parameter("Indicates name of exploration test to run. If not specified, will run all tests sequentially.")]
+    readonly ExplorationTestName ExplorationTestName;
```

Comment:
Can we use the existing `SampleName` parameter? These parameters show in the help for _all_ the targets, and we already have a _looong_ list, so I'm wary about adding new ones now! 😄

Reply:
This means, that ExplorationTestName will not be an `Enumeration`, but `string`. 🤔
And this means, that validation of this parameter should be implemented during `Target` execution.
@OmerRaviv are you OK with that?

Reply:
Ah good point, I wouldn't worry about it then, it's a minor thing really.